### PR TITLE
planner, executor: fix `PREPARE FROM @var_name`

### PR DESCRIPTION
**Pull request by AndrewDi**
### What problem does this PR solve? 
PlanBuilder: Fix prepare from session value issue #8074 

### What is changed and how it works?
When prepare from a user defined session name, replace session name with session value. 
Improve mysql grammar compatibility.

### Check List 

Tests 

 - Unit test

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8437)

### CODE REVIEW

**Comment by AndrewDi** [comment]
@XuHuaiyu @shenli  It seems like go mod dependence have some errors during CI test ...
```
go: cloud.google.com/go@v0.26.0: unrecognized import path "cloud.google.com/go" (parse https://cloud.google.com/go?go-get=1: no go-import meta tags ())
```

**Comment by AndrewDi** [comment]
Test case address...

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -396,17 +396,20 @@ func (b *PlanBuilder) buildSelectLock(src LogicalPlan, lock ast.SelectLockType)
...
+		if v, ok := b.ctx.GetSessionVars().Users[x.SQLVar.Name]; ok {
+			p.SQLText = v
+		} else {
+			p.SQLText = "NULL"
+		}
 	} else {
 		p.SQLText = x.SQLText
 	}
-	return p
+	return p, nil
```
Comment (eurekaka) [inline planner/core/planbuilder.go]: Why adding one more return value error which is always nil?

File: executor/prepared_test.go [hunk 2]
```diff
@@ -214,6 +214,15 @@ func (s *testSuite) TestPrepared(c *C) {
...
 		c.Assert(len(fields), Equals, 0)
 
+		// issue 8074
+		tk.MustExec("drop table if exists prepare1;")
+		tk.MustExec("create table prepare1 (a decimal(1))")
+		_, err = tk.Exec("prepare stmt FROM @sql1")
+		c.Assert(err.Error(), Equals, "line 1 column 4 near \"\" (total length 4)")
+		tk.MustExec("SET @sql = 'update prepare1 set a=5 where a=?';")
+		_, err = tk.Exec("prepare stmt FROM @sql")
+		c.Assert(err, IsNil)
```
[the remark this answers is not in the document]
Comment (AndrewDi) [reply executor/prepared_test.go]: fine, result check addressed.

File: planner/core/planbuilder.go (hunk 1 above)
Comment (AndrewDi) [reply planner/core/planbuilder.go]: I think we may address some errors in this function first thought; but found no such errors should be address here.

Comment (eurekaka) [reply planner/core/planbuilder.go]: Then I prefer removing it.

Comment (XuHuaiyu) [reply planner/core/planbuilder.go]: Address this comment makes an LGTM. @AndrewDi

**Comment by zz-jason** [comment]
Hi, @AndrewDi , could you cherry pick this commit to the release-2.1 and release-2.0 branch?

**Pull request merged**